lib/GADS.pm

```diff
@@ -3024,6 +3024,62 @@ prefix '/:layout_name' => sub {
 
     };
 
+    any [ 'get', 'post' ] => '/historic_purge/' => require_login sub {
+        my $layout = var('layout') or pass;
+
+        forwardHome({ danger => 'You do not have permission to access this page.' })
+            unless $layout->user_can('purge');
+
+        my $user = logged_in_user;
+        my $view = current_view($user, $layout);
+
+        my %params = (
+            user   => $user,
+            layout => $layout,
+            schema => schema,
+            view   => $view,
+        );
+
+        if (defined param('stage1'))
+        {
+            $params{columns} = [ body_parameters->get_all('column_id') ];
+
+            my $records = GADS::Records->new(%params);
+            my @columns = grep { !$_->internal } @{ $records->columns_render };
+
+            my $table_data;
+            my $ids = [];
+
+            while (my $record = $records->single)
+            {
+                my @mapped_columns =$record->presentation_map_columns(columns => \@columns);
+
+                foreach my $column (@mapped_columns)
+                {
+                    push @$ids, $column->{id};
+                    $table_data->{ $record->current_id }->{ $column->{name} } = $column->{data}->{value} || "No value";
```

Comment:
Please use field ID rather than field name, in case of duplicates. The column name can then be retrieved either from the layout object or probably better by adding to `$table_data`.